compiler.ts

```diff
@@ -10,7 +10,12 @@ export function isWholeProgramTarget(compileTarget: string) {
     return compileTarget == "METAL" || compileTarget == "SPIRV";
 }
 
-const imageMainSource = `
+export const RUNNABLE_ENTRY_POINT_NAMES = ['imageMain', 'printMain'] as const;
+export type RunnableShaderType = typeof RUNNABLE_ENTRY_POINT_NAMES[number]
```

Comment:
Should this end with a ;?

Reply:
Semicolons in js/ts are optional, but for formatting reasons they should be consistent, yes. I've gone ahead and tried to add semicolons to all the lines that they were missing on, and made the formatting overall more consistent.